mono/mini/main.c

```diff
@@ -179,8 +182,49 @@ probe_embedded (const char *program, int *ref_argc, char **ref_argv [])
 		goto doclose;
 	if (read (fd, sigbuffer, sizeof (sigbuffer)) == -1)
 		goto doclose;
-	if (memcmp (sigbuffer+sizeof(uint64_t), "xmonkeysloveplay", 16) != 0)
+	if (memcmp (sigbuffer+sizeof(uint64_t), "xmonkeysloveplay", 16) == 0)
+		goto found;
+	else
+	{
+#ifdef HOST_DARWIN
+		struct mach_header_64 h;
```

Comment:
Please use a more descriptive variable name than `h`. `header` or something is fine.

Reply:
renamed "h" to "bin_header" in https://github.com/mono/mono/pull/19580/commits/0cc053706a8539e527cdb2410902f1bf7f77d9d7